azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,494 @@
+            spc_client_id=spc_client_id,
+        )
+
+        return self._add_secrets_to_secret_sync(
+            secrets=[(secret_name, file_name)],
+            secret_sync=opcua_secret_sync,
+            resource_group=resource_group,
+            spc_name=OPCUA_SPC_NAME,
+            secret_sync_name=OPCUA_ISSUER_LIST_SECRET_SYNC_NAME,
+        )
+
+    def client_add(
+        self,
+        instance_name: str,
+        resource_group: str,
+        public_key_file: str,
+        private_key_file: str,
+        subject_name: str,
+        application_uri: str,
+    ) -> dict:
+        # inform user if the provided cert was issued by a CA, the CA cert must be added to the issuers list.
+        logger.warning("Please ensure the certificate must be added to the issuers list if it was issued by a CA. ")
+        cl_resources = self._get_cl_resources(instance_name=instance_name, resource_group=resource_group)
+        secretsync_spc = self._find_existing_spc(instance_name=instance_name, cl_resources=cl_resources)
+        if not secretsync_spc:
+            return
+
+        # get properties from default spc
+        spc_properties = secretsync_spc.get("properties", {})
+        spc_keyvault_name = spc_properties.get("keyvaultName", "")
+        spc_client_id = spc_properties.get("clientId", "")
+        spc_tenant_id = spc_properties.get("tenantId", "")
+
+        self.keyvault_client = get_keyvault_client(
+            subscription_id=self.default_subscription_id,
+            keyvault_name=spc_keyvault_name,
+        )
+
+        secrets: PageIterator = self.keyvault_client.list_properties_of_secrets()
+
+        # process all the file validations before secret creations
+        self._validate_key_files(public_key_file, private_key_file)
+
+        # check if there is a spc called "opc-ua-connector", if not create one
+        try:
+            opcua_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
+                resource_group_name=resource_group,
+                azure_key_vault_secret_provider_class_name=OPCUA_SPC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_spc = {}
+
+        # check if there is a secret sync called "aio-opc-ua-broker-client-certificate", if not create one
+        try:
+            opcua_secret_sync = self.ssc_mgmt_client.secret_syncs.get(
+                resource_group_name=resource_group,
+                secret_sync_name=OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_secret_sync = {}
+
+        secrets_to_add = []
+        for file in [public_key_file, private_key_file]:
+            file_name = os.path.basename(file)
+            file_name_info = os.path.splitext(file_name)
+            cert_extension = file_name_info[1].replace(".", "")
+            cert_name = file_name_info[0].replace(".", "-")
+            secret_name = f"{cert_name}-{cert_extension}"
```

Comment:
why does this secret name include the extension?

Reply:
because there will be cert name same but extension not same situation, so we want both part in the name